File: api/Models/BuyOrderDetail.cs
using Microsoft.EntityFrameworkCore;

namespace api.Models
{
    public class BuyOrderDetail
    {
        public int BuyOrderId { get; set; }
        public BuyOrder BuyOrder { get; set; } = default!;
        public int ProductId { get; set; }
        public Product Product { get; set; } = default!;
        public int Quantity { get; set; }
        [Precision(18, 2)]
        public decimal UnitCost { get; set; }
        [Precision(18, 2)]
        public decimal Total => Quantity * UnitCost;
    }
}

File: api/app/models/buy_order_detail.ts
import { BaseModel, belongsTo, column } from '@adonisjs/lucid/orm';
import BuyOrder from './buy_order.js';
import type { BelongsTo } from '@adonisjs/lucid/types/relations';
import Product from './product.js';

export default class BuyOrderDetail extends BaseModel {
  @column()
  declare buyOrderId: number;

  @column()
  declare productId: number;

  @column()
  declare quantity: number;

  @column()
  declare unitCost: number;

  @belongsTo(() => BuyOrder)
  declare buyOrder: BelongsTo<typeof BuyOrder>;

  @belongsTo(() => Product)
  declare product: BelongsTo<typeof Product>;
}
